pulumi/grapl/__main__.py

```diff
@@ -227,13 +227,44 @@ def main() -> None:
         # Not doing this in Local Grapl at the moment, as we have
         # another means of doing this. We should harmonize this, of
         # course.
-        ENGAGEMENT_VIEW_DIR = repository_path("src/js/engagement_view/build").resolve()
+        # If we have a version for the grapl-ux, we should have
+        # downloaded the artifact from the artifact repository and
+        # unpacked it into a version-tagged directory for us to pull
+        # from.
+        #
+        # We have to do this because there isn't an apparent way in
+        # Pulumi to expand a tarball into an S3 bucket. If that every
+        # becomes possible, we should us it, by all means!
+        #
+        # If there's no configured version, we just pull from the
+        # build directory in the local source tree (it will need to
+        # have been built first, too).
+        grapl_ux_version = configured_version_for("grapl-ux")
```

Comment:
Can we pull all this ux-specific stuff out to its own function? I think it's large and self-contained enough to make an argument against having it in main()

Reply:
Yup, that's fair. Will do 👍